crates/storage/libmdbx-rs/src/txn_manager.rs

```diff
@@ -249,13 +282,13 @@ mod read_transactions {
 
                             // Add the transaction to the list of aborted transactions, so further
                             // usages report the correct error when the transaction is closed.
-                            self.add_aborted(ptr);
-
-                            // Abort the transaction
-                            let result = mdbx_result(unsafe { ffi::mdbx_txn_abort(ptr) });
+                            let result = self
+                                .add_aborted(ptr)
+                                .and_then(|_| mdbx_result(unsafe { ffi::mdbx_txn_abort(ptr) }));
```

Comment:
not sure if this change makes sense to me, I think we need to better understand how MDBX re-uses transaction pointers to decide if that's correct.

The previous assumption was that pointers are **always** unique, so if we added a transaction to the `aborted` list, we won't be able to stumble upon it in the `active` list anymore, because we remove it from the list down below.

Reply:
this change was made since I thought abort by user should also add to aborted list